crates/storage/nippy-jar/src/cursor.rs

```diff
@@ -36,10 +35,13 @@ impl<'a> NippyJarCursor<'a> {
         config: &'a NippyJar,
         zstd_decompressors: Option<Mutex<Vec<Decompressor<'a>>>>,
     ) -> Result<Self, NippyJarError> {
+        let file = File::open(config.data_path())?;
+        let mmap = unsafe { Mmap::map(&file)? };
```

Comment:
should note that this doesn't makes any guarantees about how/when things are flushed - dirty pages can be flushed at will since we do not do any shadow paging or other sync mechanisms. if that's fine then this PR looks good to me

Reply:
this only deals with read only files, so it should be fine